Library/ViewModels/PledgeViewModelTests.swift

```diff
@@ -1423,6 +1468,83 @@ final class PledgeViewModelTests: TestCase {
     self.submitButtonEnabled.assertValues([false, false, true, false], "Amount unchanged")
   }
 
+  func testUpdatingRewardSubmitButtonEnabled_ShippingEnabled() {
+    let reward = Reward.postcards
+      |> Reward.lens.shipping.enabled .~ true
+
+    let project = Project.cosmicSurgery
+      |> Project.lens.state .~ .live
+      |> Project.lens.personalization.isBacking .~ true
+      |> Project.lens.personalization.backing .~ (
+        .template
+          |> Backing.lens.paymentSource .~ .template
+          |> Backing.lens.status .~ .pledged
+          |> Backing.lens.reward .~ Reward.otherReward
+          |> Backing.lens.rewardId .~ Reward.otherReward.id
+          |> Backing.lens.shippingAmount .~ 10
+          |> Backing.lens.amount .~ 700
+      )
+
+    self.submitButtonTitle.assertDidNotEmitValue()
+    self.submitButtonEnabled.assertDidNotEmitValue()
+
+    let updateBackingEnvelope = UpdateBackingEnvelope(
+      updateBacking: .init(
+        checkout: .init(
+          state: .successful,
+          backing: .init(
+            requiresAction: false,
+            clientSecret: "client-secret"
+          )
+        )
+      )
+    )
+
+    let mockService = MockService(
+      updateBackingResult: .success(updateBackingEnvelope)
+    )
+
+    withEnvironment(apiService: mockService, currentUser: .template) {
+      self.vm.inputs.configureWith(project: project, reward: reward, refTag: nil, context: .updateReward)
+      self.vm.inputs.viewDidLoad()
+
+      self.submitButtonTitle.assertValues(["Confirm"])
+      self.submitButtonEnabled.assertValues([false])
+      self.notifyDelegateUpdatePledgeDidSucceedWithMessage.assertDidNotEmitValue()
+      self.updatePledgeFailedWithError.assertDidNotEmitValue()
+
+      self.vm.inputs.pledgeAmountViewControllerDidUpdate(
+        with: (amount: 690, min: 25.0, max: 10_000.0, isValid: true)
+      )
+
+      self.submitButtonEnabled.assertValues([false, false], "Amount unchanged")
```

Comment:
I might be missing something with this test, but why is the submit button disabled here? I would expect that because the VM is being configured with `reward` (which is different from `project.backing.reward`) that the submit button would be enabled because the pledge amount is between the min and max _of the new reward_. What am I missing here?

Reply:
The reward has shipping enabled so the way the validation rules are working is that a shipping rule needs to be selected in order to change the reward.